### PR TITLE
Reorganize internal export code.

### LINKED ISSUE #4107: Move GrpcExporter (internal) class out of "otlp" (internal) package
Just to avoid confusion like in #4106 that makes someone think they are using the otlp exporter when they're not really.

### PR DESCRIPTION
**Pull request by anuraaga**
`exporter.otlp.internal` -> `exporter.otlp.internal`

and moves non-otlp packages under `exporter.otlp`

This doesn't solve the fact that the artifact is still called otlp-common but that should have less chance of confusing errors vs error messages that contain the package names, and it will be easier to split out an artifact if desired later.

Fixes #4107

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerUtil.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
     JSON_AVAILABLE = jsonAvailable;
   }
 
-  static final byte[] EMPTY_BYTES = new byte[0];
+  public static final byte[] EMPTY_BYTES = new byte[0];
```
Comment (jack-berg) [inline exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerUtil.java:49]: Consider adding a method to access this.

**Review by jack-berg — APPROVED** [review]

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Serializer.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public void serializeBool(ProtoFieldInfo field, boolean value) throws IOExceptio
     writeBool(field, value);
   }
 
-  protected abstract void writeBool(ProtoFieldInfo field, boolean value) throws IOException;
+  public abstract void writeBool(ProtoFieldInfo field, boolean value) throws IOException;
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Serializer.java:56]: still internal. do we need javadoc on internal methods, even if public?

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerUtil.java (hunk 1 above)
Comment (anuraaga) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerUtil.java]: Just made it private instead

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Serializer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jack-berg) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Serializer.java:57]: I say yes, but hold it to the same standard as I would non-internal methods.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**